src/panels/ConfigPanel/ConfigPanel.tsx

```diff
@@ -215,29 +220,62 @@ export default function ConfigPanel(): JSX.Element {
   function generateConfirmationDialog(): JSX.Element | null {
     if (confirmType && appName) {
       const changedConfigList = generateChangedKeyValues(config);
-
-      if (confirmType === "apply") {
-        return SaveConfirmation(
-          appName,
-          changedConfigList,
-          () => {
-            setConfirmType(null);
-            // Clear the form errors if there were any from a previous submit.
-            setFormErrors(null);
-            _submitToJuju();
-          },
-          () => setConfirmType(null)
+      if (confirmType === "submit") {
+        // Render the submit confirmation modal.
+        return (
+          <Portal>
+            <ConfirmationModal
+              // Prevent clicks inside this panel from closing the parent panel.
+              // This is handled in `checkCanClose`.
+              className="prevent-panel-close"
+              title={Label.SAVE_CONFIRM}
+              cancelButtonLabel={Label.SAVE_CONFIRM_CANCEL_BUTTON}
+              confirmButtonLabel={Label.SAVE_CONFIRM_CONFIRM_BUTTON}
+              confirmButtonAppearance="positive"
+              onConfirm={() => {
+                setConfirmType(null);
+                // Clear the form errors if there were any from a previous submit.
+                setFormErrors(null);
+                _submitToJuju();
+              }}
+              close={() => setConfirmType(null)}
+            >
+              <p>
+                You have edited the following values to the {appName}{" "}
+                configuration:
+              </p>
+              {changedConfigList}
+              <div className="config-panel__modal-button-row-hint">
+                You can revert back to the applications default settings by
+                clicking the “Reset all values” button; or reset each edited
+                field by clicking “Use default”.
+              </div>
```

Comment:
To make this look a little like the old confirmation, you could pass this as a prop to `<ConfirmationModal>`:

```

              confirmExtra={
                <p className="u-text--muted u-align--left">
                  You can revert back to the applications default settings by
                  clicking the “Reset all values” button; or reset each edited
                  field by clicking “Use default”.
                </p>
              }
```

The `&__modal-button-row-hint` class can then be removed from `_config-panel.scss`.